### PR TITLE
Update deb packaging to work with bionic (18.04)

### PR DESCRIPTION
**Pull request by rosswhitfield**
Installing the Ubuntu 18.04 package [currently fails](http://builds.mantidproject.org/job/master_systemtests-ubuntu-18.04-python3/1/console). This fixes that.

This also removes support for packaging trusty (14.04)

**To test:**
Build package on Ubuntu 18.04 and see if it now installs. Check that the 16.04 package still works.
Check both python 2 and 3 packages.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by AndreiSavici — CHANGES_REQUESTED** [review]
Built for python3 on my workstation (Ubuntu 18.04), with documentation packed. Installed on my laptop after a clean Ubuntu 18.04 install. 

```
andrei@andrei-Aspire-VN7-791:/opt/mantidunstable/bin$ ./launch_mantidplot.sh 
/opt/mantidunstable/bin/MantidPlot_exe: error while loading shared libraries: libQtWebKit.so.4: cannot open shared object file: No such file or directory
```

After installing libqtwebkit manually, I can import mantd.simpleapi in python. However, I cannot start mantidplot

```
andrei@andrei-Aspire-VN7-791:/opt/mantidunstable$ bin/launch_mantidplot.sh 
/opt/mantidunstable/bin/MantidPlot_exe: error while loading shared libraries: libMantidDataObjects.so: cannot open shared object file: No such file or directory
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -353,17 +353,18 @@ if ( ENABLE_CPACK )
...
                            "libpocofoundation${POCO_SOLIB_VERSION},libpocoutil${POCO_SOLIB_VERSION},libpoconet${POCO_SOLIB_VERSION},libpoconetssl${POCO_SOLIB_VERSION},libpococrypto${POCO_SOLIB_VERSION},libpocoxml${POCO_SOLIB_VERSION},"
+                           "libhdf5-cpp-11,"
                            "python-pycifrw (>= 4.2.1),"
                            "python-yaml,"
                            "python-qtawesome")
         set ( PERFTOOLS_DEB_PACKAGE "libgoogle-perftools4 (>= 1.7)" )
-        if( "${UNIX_CODENAME}" STREQUAL "xenial")
-            list ( APPEND DEPENDS_LIST ", libhdf5-cpp-11,libnexus0v5 (>= 4.3),libjsoncpp1,libqscintilla2-12v5, libmuparser2v5,libqwtplot3d-qt4-0v5,libgsl2,liboce-foundation10,liboce-modeling10")
-            list (REMOVE_ITEM DEPENDS_LIST "libjsoncpp0 (>=0.7.0)," "libqscintilla2-11," "libmuparser2," "libnexus0 (>= 4.3)," "libqwtplot3d-qt4-0," "libgsl0ldbl," "liboce-foundation8,liboce-modeling8,")
+        if( "${UNIX_CODENAME}" STREQUAL "bionic")
```
Comment (martyngigg) [inline CMakeLists.txt]: Are we not keeping the Xenial packaging requirements? ILL will need this I think...

Comment (rosswhitfield) [reply CMakeLists.txt]: Xenial is now the default list, instead of trusty. Then we just change the dependancy list for bionic.

**Review by AndreiSavici — APPROVED** [review]

**Comment by AndreiSavici** [comment]
It's a step in the right direction.

**Review by martyngigg — APPROVED** [review]
After a small fix for a missing comma in the dependency list this works on Ubuntu 16.04

**Pull request merged**